vignettes/bcva.Rmd.upstream

```diff
@@ -0,0 +1,391 @@
+head(bcva_data) |>
+  gt() |>
+  tab_caption(caption = md("Table 1. First rows of the pre-processed `bcva_data` dataset."))
+```
+
+## Descriptive statistics
+
+Table of baseline characteristics:
+
+```{r descr_1, eval=TRUE, echo=TRUE, class.source = 'fold-hide'}
+bcva_data |>
+  select(ARMCD, USUBJID, RACE, BCVA_BL) |>
+  distinct() |>
+  select(-USUBJID) |>
+  tbl_summary(
+    by = c(ARMCD),
+    statistic = list(
+      all_continuous() ~ "{mean} ({sd})",
+      all_categorical() ~ "{n} / {N} ({p}%)"
+    )
+  ) |>
+  modify_caption("Table 2. Baseline characteristics.")
+```
+
+Table of change from baseline in BCVA over 52 weeks:
+
+```{r descr_2, eval=TRUE, echo=TRUE, class.source = 'fold-hide'}
+bcva_data |>
+  pull(AVISIT) |>
+  unique() |>
+  sort() |>
+  purrr::map(
+    .f = ~ bcva_data |>
+      filter(AVISIT %in% .x) |>
+      tbl_summary(
+        by = ARMCD,
+        include = BCVA_CHG,
+        type = BCVA_CHG ~ "continuous2",
+        statistic = BCVA_CHG ~ c(
+          "{mean} ({sd})",
+          "{median} ({p25}, {p75})",
+          "{min}, {max}"
+        ),
+        label = list(BCVA_CHG = paste("Visit ", .x))
+      )
+  ) |>
+  tbl_stack(quiet = TRUE) |>
+  modify_caption("Table 3. Change from baseline.")
+```
+
+The following figure shows the primary endpoint over the four study visits in the data.
+
+```{r descr_3, eval=TRUE, echo=TRUE, fig.align='center', out.width="75%", fig.dim = c(8, 5), fig.fig.keep='all', fig.pos="!ht", fig.cap="Figure 1. Change from baseline in BCVA over 4 visit time points."}
+bcva_data |>
+  group_by(ARMCD) |>
+  ggplot(aes(x = AVISIT, y = BCVA_CHG, fill = factor(ARMCD))) +
+  geom_hline(yintercept = 0, col = "grey", linewidth = 1.2) +
+  geom_boxplot(na.rm = TRUE) +
+  labs(
+    x = "Visit",
+    y = "Change from baseline in BCVA",
+    fill = "Treatment"
+  ) +
+  scale_fill_manual(values = c("darkgoldenrod2", "coral2")) +
+  theme_bw()
+```
+
+# Fitting MMRMs
+
+## Bayesian model
+
+The formula for the Bayesian model includes additive effects for baseline, study visit, race, and study-arm-by-visit interaction.
+
+```{r bayes_mmrm_1, eval=TRUE, echo=TRUE, class.source = 'fold-hide'}
+b_mmrm_formula <- brm_formula(
+  data = bcva_data,
+  intercept = TRUE,
+  baseline = TRUE,
+  group = FALSE,
+  time = TRUE,
+  baseline_time = FALSE,
+  group_time = TRUE,
+  correlation = "unstructured"
+)
+print(b_mmrm_formula)
+```
+
+We fit the model using `brms.mmrm::brm_model()`. To ensure a good basis of comparison with the frequentist model, we put an extremely diffuse prior on the intercept. The parameters already have diffuse flexible priors by default.
+
+```{r bayes_mmrm_2b, eval=TRUE, echo=TRUE, cache=FALSE, class.source = 'fold-hide'}
+b_mmrm_fit <- brm_model(
+  data = filter(bcva_data, !is.na(BCVA_CHG)),
+  formula = b_mmrm_formula,
+  prior = brms::prior(class = "Intercept", prior = "student_t(3, 0, 1000)"),
+  iter = 10000,
+  warmup = 2000,
+  chains = 4,
+  cores = 4,
```

Comment:
`cores = 4` may not be CRAN compatible, if the chunk is actually supposed to run when building the vignette (as far as I am aware the limit is 2 cores). You are probably aware anyways.

Reply:
I thought about this, and because the `bcva.Rmd` is pre-rendered from `bcva.Rmd.upstream`, it seems unlikely that CRAN will notice or care. I used the technique from https://ropensci.org/blog/2019/12/08/precompute-vignettes/. Is that okay with you?

Reply:
Yes, pre-rendered sounds like a good idea in this case.